src/script.ts

```diff
@@ -110,45 +122,190 @@ export const runScript = async (
+    throw new Error('Failed to parse script content')
+  }
+
+  let browserImport: ts.ImportDeclaration | null = null
+  let httpImport: ts.ImportDeclaration | null = null
+
+  traverse(scriptAst, {
+    [NodeType.ImportDeclaration](node) {
+      switch (node.source.value) {
+        case 'k6/http':
+          httpImport = node
+          break
+
+        case 'k6/browser':
+          browserImport = node
+          break
+      }
+    },
+  })
+
+  if (httpImport !== null) {
+    // Insert the group shim right after the http import.
+    scriptAst.body = scriptAst.body
+      .filter((statement) => !isConflictingExecutionImport(statement))
+      .flatMap((statement) =>
+        statement === httpImport ? [httpImport, ...groupAst.body] : statement
+      )
+  }
+
+  const exports = getExports(scriptAst)
+
+  const hasHandleSummary = exports.some(
+    (e) => e.type === 'named' && e.name === 'handleSummary'
   )
 
-  // NOTE: checks works only if the user doesn't define a custom summary handler
-  // if no custom handleSummary is defined we add our version to retrieve checks
-  if (handleSummaryIndex === -1) {
-    scriptLines.push(checksSnippet)
+  if (!hasHandleSummary) {
+    scriptAst.body.push(...checksAst.body)
   }
 
-  if (httpImportIndex !== -1) {
-    scriptLines.splice(httpImportIndex + 1, 0, groupSnippet)
-    const modifiedScriptContent = scriptLines.join('\n')
-    return modifiedScriptContent
-  } else {
-    return scriptLines.join('\n')
+  // Find any existing options export and replace it with our own.
+  const optionsExport = exports.find(
+    (e) => e.type === 'named' && e.name === 'options'
+  )
+
+  if (optionsExport) {
+    traverse(scriptAst, {
+      [NodeType.Identifier](node) {
+        if (node.name === 'options') {
+          // It's easier to just rename the options export than to
+          // remove the node itself. This is just some UUID that I
+          // generated. Should be pretty unique.
+          node.name = '$c26e2908c2e948ef883369abc050ce2f'
+        }
+      },
+    })
   }
-}
 
-const getJsSnippet = async (snippetName: string) => {
-  let jsSnippetPath: string
+  const browserOptions =
+    browserImport !== null
+      ? {
+          options: fromObjectLiteral({
+            browser: fromObjectLiteral({
+              type: 'chromium',
+            }),
+          }),
+        }
+      : null
 
-  // if we are in dev server we take resources directly, otherwise look in the app resources folder.
+  const options = fromObjectLiteral({
+    scenarios: fromObjectLiteral({
+      default: fromObjectLiteral({
+        executor: 'shared-iterations',
+        vus: 1,
+        iterations: 1,
+        ...browserOptions,
+      }),
+    }),
+  })
+
+  scriptAst.body.push(
+    exportNamed({
+      declaration: declareConst({
+        declarations: [
+          constDeclarator({
+            id: identifier('options'),
+            init: options,
+          }),
+        ],
+      }),
+    })
+  )
```

Comment:
A more sophisticated version would try to merge the injected options with existing options, but I think this a good enough start.